### ISSUE TITLE
Bulk resize doesn't resize images on S3

### ISSUE THREAD

**Issue by billmn** [engram-design/ImageResizer#59]
Hi guys,
I have a volume with S3 in which I have already uploaded quite large images.

Then I installed the plugin and performed the "Bulk Resize" in order to resize the existing images. The problem is that it apparently did not resize any image but only update database fields related to its size.

If I delete all files and re-upload, images are resized correctly. It therefore appears that the problem may be related to previously existing images. 

I am attaching some screenshots in the hope that they will be useful:

## Assets

BEFORE

![Schermata 2021-02-26 alle 10 36 33](https://user-images.githubusercontent.com/779534/109284227-ecf91c00-781f-11eb-9808-7fff95e9d1d6.png)

AFTER

![Schermata 2021-02-26 alle 10 41 03](https://user-images.githubusercontent.com/779534/109284522-3c3f4c80-7820-11eb-927b-d7d2f77b1799.png)

## AWS S3

BEFORE

![Schermata 2021-02-26 alle 10 39 04](https://user-images.githubusercontent.com/779534/109285000-d0111880-7820-11eb-8d41-64e2d6565725.png)

AFTER (it's not the wrong screenshot, it looks like the files have not been touched)

![Schermata 2021-02-26 alle 10 49 50](https://user-images.githubusercontent.com/779534/109285018-d606f980-7820-11eb-9758-c559a478e519.png)

## Dialog and Log

![Schermata 2021-02-26 alle 10 40 26](https://user-images.githubusercontent.com/779534/109284481-3184b780-7820-11eb-9b2b-7395be54ef85.png)

![Schermata 2021-02-26 alle 11 04 05](https://user-images.githubusercontent.com/779534/109286311-5e39ce80-7822-11eb-9020-2cab9cbaadf4.png)

## Configuration

![Schermata 2021-02-26 alle 10 39 46](https://user-images.githubusercontent.com/779534/109285396-4f9ee780-7821-11eb-9d3d-43e8a1d81174.png)

**Steps to reproduce**
1. Create a S3 volume and upload large images
2. Install `Image Resizer` plugin
3. Check image's file size and dimensions in Assets section
4. Run Bulk Resize
5. Check image's file size and dimensions again

**Additional info**
- Plugin version: 2.1.0
- Craft version: 3.6.7
- Multi-site: Yes

**Comment by alexroper** [comment]
I can confirm this bug also. After running the bulk resize command, I can locate the images in `craft/storage/runtime/assets/sources` and they have been resized properly, but the image source on Amazon S3 has not changed.

Craft Pro: 3.7.24
Image Resizer: 2.1.0
Amazon S3: 1.3.0
PHP: 7.4.22

**Comment by eeeeeeeeeve** [comment]
@engram-design FYI this seems to be because the image gets saved to /storage/runtime/assets/sources, but Craft doesn't allow this:

```
Prevented saving [...]/storage/runtime/assets/sources/REDACTED.jpg as an asset. It must be located within a temp directory or the project root (excluding system directories).
```

On [this line](https://github.com/verbb/image-resizer/blob/craft-4/src/jobs/ImageResize.php#L55), it uses `getImageTransformSourcePath()` for a pre-existing file on S3 which seems to end up in the local filesystem as `storage/runtime/assets/sources`, which is disallowed to be saved because it's in the runtime path.

Fix would be to save this elsewhere (not a system path) and clean it up afterwards.

Refs:
- https://github.com/craftcms/cms/blob/7447808979ecd27a92857ade64335fd382215921/src/elements/Asset.php#L3288-L3299
- https://github.com/craftcms/cms/blob/7447808979ecd27a92857ade64335fd382215921/src/services/Path.php#L585

**Comment by engram-design** [comment]
I can't seem to replicate that error, no matter what I do I'm afraid. It's true that if no local file exists from S3, it will download a local copy of it to resize, then save it against the assets (which uploads/replaces it). I believe the [code](https://github.com/craftcms/cms/blob/7447808979ecd27a92857ade64335fd382215921/src/elements/Asset.php#L3288-L3299) you're referring to is only fired when relocating a file, not editing existing ones.

However, the original issue should be fixed, where the details in remote providers weren't being reflected by their local Craft details on size. Fixed for the next release. To get this early, run `composer require verbb/image-resizer:"dev-craft-4 as 3.0.10"`.

**Comment by engram-design** [comment]
Fixed in [3.0.11](https://github.com/verbb/image-resizer/releases/tag/3.0.11)
